### ISSUE TITLE
Subchart does not sees installation when first upgrade is run

### ISSUE THREAD

**Issue by GioPat** [helm/helm#12404]
Output of `helm version`:
```
version.BuildInfo{Version:"v3.11.2", GitCommit:"912ebc1cd10d38d340f048efaf0abda047c3468e", GitTreeState:"clean", GoVersion:"go1.18.10"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.3", GitCommit:"9e644106593f3f4aa98f8a84b23db5fa378900bd", GitTreeState:"clean", BuildDate:"2023-03-15T13:40:17Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"windows/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25+", GitVersion:"v1.25.12-eks-2d98532", GitCommit:"0aa16cf4fac4da27b9e9e9ba570b990867f6a3d8", GitTreeState:"clean", BuildDate:"2023-07-28T16:52:04Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): EKS

This is the context: I have an Umbrella chart that has lot of revisions (>100) so I keep updating it using `helm upgrade` command.
At one point I need to add a dependency to the [Rabbit MQ Bitnami chart](https://github.com/bitnami/charts/tree/main/bitnami/rabbitmq), the issue is that if I do `helm upgrade` as always, the subchart "sees" the update command instead of the install and I receive the following error message
```
Error: UPGRADE FAILED: execution error at (chart-umbrella/charts/rabbitmq/templates/statefulset.yaml:39:28):
PASSWORDS ERROR: You must provide your current passwords when upgrading the release.
                 Note that even after reinstallation, old credentials may be needed as they may be kept in persistent volume claims.
                 Further information can be obtained at https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases

    'auth.erlangCookie' must not be empty, please add '--set auth.erlangCookie=$RABBITMQ_ERLANG_COOKIE' to the command. To get the current value:

        export RABBITMQ_ERLANG_COOKIE=$(kubectl get secret --namespace "prod" chart-umbrella-rabbitmq -o jsonpath="{.data.rabbitmq-erlang-cookie}" | base64 -d)
```
To summarize: the first time I'm upgrading and I need to install the subchart instead of upgrading it, the sub chart sees an upgrade instead of a first installation.
I'd like to avoid installing manually the subchart
